app.py

```diff
@@ -38,29 +42,32 @@ def api_available():
 
 @app.route("/api/submit/criteria", methods=["POST"])
 def api_submit_criteria():
-    global criteria
+    global result, criteria
     # Raises 400 error if th request body is not a valid json
     data = request.get_json()
 
-    # TODO: data validation
-    if False:
+    if 'data' not in data.keys() or type(data['data']) != list:
```

Comment:
```suggestion
    if "data" not in data or type(data["data"]) != list:
```
No need for the `.keys()` method.
Also, let's use double or single quotes consequently throughout the file.